pkg/build/api/validation/validation.go

```diff
@@ -445,6 +465,35 @@ func validateCustomStrategy(strategy *buildapi.CustomBuildStrategy, fldPath *fie
 	return allErrs
 }
 
+func validateJenkinsPipelineStrategy(strategy *buildapi.JenkinsPipelineBuildStrategy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(strategy.JenkinsfilePath) != 0 && len(strategy.Jenkinsfile) != 0 {
+		return append(allErrs, field.Invalid(fldPath, strategy, "must provide a value for at most one of jenkinsfilePath, or jenkinsfile"))
+	}
+
+	if len(strategy.Jenkinsfile) > maxJenkinsfileLengthBytes {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("jenkinsfile"), "", fmt.Sprintf("must be smaller than %d bytes", maxJenkinsfileLengthBytes)))
+	}
+
+	if len(strategy.JenkinsfilePath) != 0 {
+		cleaned := path.Clean(strategy.JenkinsfilePath)
+		switch {
+		case filepath.IsAbs(cleaned):
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("jenkinsfilePath"), strategy.JenkinsfilePath, "jenkinsfilePath must not be an absolute path"))
```

Comment:
Would invert the negation in the message - "jenkinsfilePath must be a relative path in your source location"